plugins/net_plugin/net_plugin.cpp

```diff
@@ -1636,8 +1644,10 @@ namespace eosio {
       peer_block_state pbstate{bs->id, bnum};
 
       std::shared_ptr<std::vector<char>> send_buffer;
-      for( auto& cp : my_impl->connections ) {
-         if( skips.find( cp ) != skips.end() || !cp->current() || cp->is_gray || cp->considers_gray ) {
+      auto itr = my_impl->connections.lower_bound(connection_state::connected);
+      for (; itr != my_impl->connections.end() && (*itr)->state == connection_state::connected; ++itr) {
+         auto& cp = *itr;
+         if (skips.find(cp) != skips.end() || !cp->current() || cp->is_gray || cp->considers_gray ) {
```

Comment:
`cp->current()` looks surplus, because the for-loop already checks state on `connected`